### PR TITLE
chart: rename backendStoreDriver to dataEngine

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#7095

### CODE REVIEW

**Comment by derekbit** [comment]
longhorn-manager part is merged. The chart update is ready to merge as well.

**Review by PhanLe1010 — DISMISSED** [review]
in general, LGTM

File: examples/v2/storageclass.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ parameters:
   numberOfReplicas: "3"
   staleReplicaTimeout: "2880"
   fsType: "ext4"
-  backendStoreDriver: "v2"
```
Comment (PhanLe1010) [inline examples/v2/storageclass.yaml:14]: For user who still have the old SC with `backendStoreDriver` field, will Longhorn webhook/CSI plugin automatically translate it to `dataEngine` ?

File: chart/templates/crds.yaml [hunk 2]
```diff
@@ -1315,12 +1319,15 @@ spec:
               active:
                 type: boolean
               backendStoreDriver:
+                description: Deprecated.
```
Comment (c3y1huang) [inline chart/templates/crds.yaml:1322]:
```suggestion
                description: 'Deprecated: Replaced by field `dataEngine`.'
```

File: examples/v2/storageclass.yaml (hunk 1 above)
Comment (derekbit) [reply examples/v2/storageclass.yaml:14]: No, users need to use `dataEngine`. I will mention it in doc.
Do you think we need to help the translation?

Comment (PhanLe1010) [reply examples/v2/storageclass.yaml:14]: I mean for users who already have the old SC with backendStoreDriver, they might not be aware of the new change. I am ok with either:
1. A release note
2. Auto translation

**Review by innobead — DISMISSED** [review]

File: chart/templates/crds.yaml (hunk 2 above)
Comment (derekbit) [reply chart/templates/crds.yaml]: Updated. Thank you.

**Review by c3y1huang — APPROVED** [review]

**Review by innobead — APPROVED** [review]

File: examples/v2/storageclass.yaml (hunk 1 above)
Comment (derekbit) [reply examples/v2/storageclass.yaml:14]: `Auto translation` is good idea for existing storageclass, but users still need to re-created their stgorageclass using `backendStoreDriver` after we remove it in v1.7.0. @innobead Do you think we should help the translation, or just let user recreate directly?

Comment (innobead) [reply examples/v2/storageclass.yaml:14]: As you mentioned, it's a preview (alpha) feature, so things could be changed. Let's have a note in the [import notes](https://longhorn.io/docs/1.6.0/deploy/important-notes/), good enough for now.

**Pull request merged**